### ISSUE TITLE
Create loader schemas for validating `kind.yml` files

### ISSUE THREAD

**Issue by ahal** [taskcluster/taskgraph#148]
The format of the `kind.yml` files is currently determined by loaders. We should provide schemas for what these loaders expect and validate `kind.yml` files against them. This would result in clearer expectations about what is allowed in these files.

This came from a request from :asuth who is looking into ways of integrating Taskgraph with Searchfox. He'd prefer the schema follow the JSON-schema spec. I'm not sure I'd want two separate schema validation methods being used (`jsonschema` + `voluptuous`), but maybe we could have a CI step to export the schema into the JSON-schema format or something.

**Comment by ahal** [comment]
Fixing this bug will involve creating a new [voluptuous](https://github.com/alecthomas/voluptuous) schema ([like we have for transforms](https://github.com/taskcluster/taskgraph/blob/main/src/taskgraph/transforms/notify.py#L56)), here:
https://github.com/taskcluster/taskgraph/blob/main/src/taskgraph/loader/transform.py

Then somewhere around [here in generator.py](https://github.com/taskcluster/taskgraph/blob/389ef97a27259ec314dfa5c019ffc203cf033d87/src/taskgraph/generator.py#L101), we'd evaluate the contents of the `kind.yml` against said schema. Possibly we could fetch the schema by looking for a common global variable in the loader module, or we could invent some more complex way of registering loaders..

**Comment by Aakash017** [comment]
Hello @ahal I would like to work on this

**Comment by Aakash017** [comment]
@ahal As per my understanding we have to create new schema using [voluptuous](https://github.com/alecthomas/voluptuous) and then evaluate all kind.yml files

**Comment by ahal** [comment]
Yes, that's correct! Though we'd only evaluate them as they are being loaded (we don't want to evaluate them all in row or anything like that, as they aren't always all needed).
